internal/common/schema/pagination_schema_test.go

```diff
@@ -0,0 +1,63 @@
+package schema_test
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	tfschema "github.com/mongodb/terraform-provider-mongodbatlas/internal/common/schema"
+)
+
+func TestPaginatedDSSchema(t *testing.T) {
+	expectedSchema := schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"project_id": schema.StringAttribute{
+				Required: true,
+			},
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"page_num": schema.Int64Attribute{
+				Optional: true,
+			},
+			"items_per_page": schema.Int64Attribute{
+				Optional: true,
+			},
+			"total_count": schema.Int64Attribute{
+				Computed: true,
+			},
+			"results": schema.ListNestedAttribute{
+				Computed: true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Computed: true,
+						},
+						"instance_name": schema.StringAttribute{
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	resultSchema := tfschema.PaginatedDSSchema(
+		map[string]schema.Attribute{
+			"project_id": schema.StringAttribute{
+				Required: true,
+			},
+		},
+		map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"instance_name": schema.StringAttribute{
+				Computed: true,
+			},
+		})
+
+	if !reflect.DeepEqual(resultSchema, expectedSchema) {
+		t.Errorf("created schema does not matched expected")
```

Comment:
knitpit: it's normally a good idea to show expected and received values when error to help understand the problem although they might be a bit long here